both/methods.js

```diff
@@ -153,16 +153,25 @@ Meteor.methods({
 
     Meteor.call('refreshRoundCodes', competitionId, eventCode);
   },
-  addNonEventRound: function(competitionId, round) {
+  addScheduleEvent: function(competitionId, eventData, roundId) {
     check(competitionId, String);
     throwIfCannotManageCompetition(this.userId, competitionId);
-    return Rounds.insert({
+
+    var round = Rounds.findOne(roundId);
+
+    return ScheduleEvents.insert({
       competitionId: competitionId,
-      title: round.title,
-      startMinutes: round.startMinutes,
-      durationMinutes: round.durationMinutes,
+      roundId: (round ? round._id : null),
```

Comment:
Hmm... this is kind of subtle behavior. If the user passes an invalid `roundId` we silently go ahead and create a `ScheduleEvent` with no `roundId`? I feel like it would be preferable to throw an error instead (unless they legitimately passed a null `roundId`). What do you think?